core/src/main/java/com/google/googlejavaformat/java/Main.java

```diff
@@ -142,28 +143,45 @@ private int formatFiles(CommandLineOptions parameters, JavaFormatterOptions opti
       } catch (ExecutionException e) {
         if (e.getCause() instanceof FormatterException) {
           for (FormatterDiagnostic diagnostic : ((FormatterException) e.getCause()).diagnostics()) {
-            errWriter.println(result.getKey() + ":" + diagnostic.toString());
+            errWriter.println(path + ":" + diagnostic.toString());
           }
         } else {
-          errWriter.println(result.getKey() + ": error: " + e.getCause().getMessage());
+          errWriter.println(path + ": error: " + e.getCause().getMessage());
           e.getCause().printStackTrace(errWriter);
         }
         allOk = false;
         continue;
       }
+      boolean idempotent = formatted.equals(inputs.get(path)); // not changed
+      if (parameters.isValidationMode()) {
+        if (idempotent) {
+          continue; // no formatting needed, validate next
+        }
+        errWriter.println(path + ": needs formatting");
+        if (parameters.isSetExitIfChanged()) {
+          allOk = false;
+        }
+        continue;
+      }
       if (parameters.inPlace()) {
-        if (formatted.equals(inputs.get(result.getKey()))) {
+        if (idempotent) {
           continue; // preserve original file
         }
         try {
-          Files.write(result.getKey(), formatted.getBytes(UTF_8));
+          Files.write(path, formatted.getBytes(UTF_8));
         } catch (IOException e) {
-          errWriter.println(result.getKey() + ": could not write file: " + e.getMessage());
+          errWriter.println(path + ": could not write file: " + e.getMessage());
           allOk = false;
           continue;
         }
-      } else {
-        outWriter.write(formatted);
+        if (parameters.isSetExitIfChanged()) {
+          allOk = false;
+        }
+        continue;
+      }
+      outWriter.write(formatted);
+      if (parameters.isSetExitIfChanged() && !idempotent) {
+        allOk = false;
       }
     }
     return allOk ? 0 : 1;
```

Comment:
I expected to see changes to formatStdin too?

Reply:
Not sure about this one.

Should an exit code of 1 be returned if changes where made for input from stdin and `--set-exit-if-changed` is set to true?

Reply:
You are referring to the method `private int formatStdin(CommandLineOptions parameters, JavaFormatterOptions options) {...}`!

Reply:
I would expect `-n` and `--set-exit-if-changed` to work the same for input from stdin as it does for files. For `google-java-format -n -` we can reuse `STDIN_FILENAME`.

Reply:
Ok. On it.

Reply:
Done via:
```java
      if (parameters.isDryRun()) {
        outWriter.println(STDIN_FILENAME);
      } else {
        outWriter.write(output);
      }
      if (input.equals(output)) {
        return 0;
      }
      return parameters.isSetExitIfChanged() ? 1 : 0;
```